### PR TITLE
feat(cart): add cart item price selectors

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Adds selectors for cart item total discount, price, quantity, and row total.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/cart/src/models/cart-item.ts [hunk 1]
```diff
@@ -20,7 +20,13 @@ export interface DaffCartItem {
   sku: string;
   name: string;
   qty: number;
+  /**
+   * The price per item, not including cart discounts.
```
Comment (lderrickable) [inline libs/cart/src/models/cart-item.ts:24]: I think we should mention that this includes category/product discounts

File: libs/cart/src/models/cart-item.ts [hunk 2]
```diff
@@ -20,7 +20,13 @@ export interface DaffCartItem {
...
+   */
   price: number;
+  /**
+   * The total for the entire row, not including cart discounts.
```
Comment (lderrickable) [inline libs/cart/src/models/cart-item.ts:28]: I think we should mention that this includes category/product discounts

File: libs/cart/state/src/facades/cart/cart-facade.interface.ts [hunk 3]
```diff
@@ -232,7 +232,31 @@ export interface DaffCartFacadeInterface<
...
+   * Zero by default.
+   */
+	getCartItemQuantity(itemId: U['item_id']): Observable<number>;
+  /**
+   * Selects the specified item's row total.
+   * Zero by default.
+   */
+	getCartItemRowTotal(itemId: U['item_id']): Observable<number>;
+  /**
+   * Selects the specified item's array of discounts.
```
Comment (lderrickable) [inline libs/cart/state/src/facades/cart/cart-facade.interface.ts:254]: I think this should specify that these are cart discounts

File: libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts [hunk 4]
```diff
@@ -135,13 +159,34 @@ const createCartItemEntitiesSelectors = <
...
+	);
+
+  const selectCartItemDiscounts = createSelector(
+		selectCartItemEntities,
+		(cartItems, props) => selectCartItem.projector(cartItems, { id: props.id })?.discounts || []
+	);
+
+  const selectCartItemTotalDiscount = createSelector(
+		selectCartItemEntities,
+		(cartItems, props) => selectCartItemDiscounts.projector(cartItems, { id: props.id })?.reduce((acc, {amount}) => acc + amount, 0) || 0
```
Comment (lderrickable) [inline libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts:187]: This should use the `daffAdd` function

File: libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts [hunk 5]
```diff
@@ -21,6 +21,30 @@ export interface DaffCartItemEntitiesMemoizedSelectors<T extends DaffStatefulCar
 	selectIsCartItemOutOfStock: MemoizedSelectorWithProps<object, object, boolean>;
 	selectCartItemMutating: MemoizedSelector<object, boolean>;
 	selectCartItemState: MemoizedSelectorWithProps<object, object, DaffCartItemStateEnum>;
+  /**
```
Comment (lderrickable) [inline libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts:24]: I think we should also mention in these comments all the stuff about including category/product discounts and excluding cart discounts, or whatever they should be

**Review by lderrickable — CHANGES_REQUESTED** [review]

File: libs/cart/src/models/cart-item.ts (hunk 1 above)
Comment (griest024) [reply libs/cart/src/models/cart-item.ts]: changed

File: libs/cart/src/models/cart-item.ts (hunk 2 above)
Comment (griest024) [reply libs/cart/src/models/cart-item.ts:29]: changed

File: libs/cart/state/src/facades/cart/cart-facade.interface.ts (hunk 3 above)
Comment (griest024) [reply libs/cart/state/src/facades/cart/cart-facade.interface.ts]: changed

File: libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts (hunk 4 above)
Comment (griest024) [reply libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts]: changed

File: libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts (hunk 5 above)
Comment (griest024) [reply libs/cart/state/src/selectors/cart-item-entities/cart-item-entities.selectors.ts:25]: changed

File: libs/cart/src/models/cart-item.ts (hunk 2 above)
Comment (lderrickable) [reply libs/cart/src/models/cart-item.ts:29]: *apply

Comment (griest024) [reply libs/cart/src/models/cart-item.ts:29]: changed

**Review by lderrickable — APPROVED** [review]

**Pull request merged**